core/src/main/java/com/sequenceiq/cloudbreak/service/StackCommonService.java

```diff
@@ -216,18 +220,33 @@ public FlowIdentifier syncInWorkspace(NameOrCrn nameOrCrn, Long workspaceId) {
     public FlowIdentifier syncComponentVersionsFromCmInWorkspace(NameOrCrn nameOrCrn, Long workspaceId, Set<String> candidateImageUuids) {
         Stack stack = stackService.getByNameOrCrnInWorkspace(nameOrCrn, workspaceId);
         MDCBuilder.buildMdcContext(stack);
-        if (stack.getStackStatus().getStatus().isStopState()) {
-            String message = String.format("Reading CM and parcel versions from CM cannot be initiated as the cluster is in %s state",
-                    stack.getStackStatus().getStatus());
-            LOGGER.debug(message);
-            throw new BadRequestException(message);
-        }
+        String operationDescription = "Reading CM and parcel versions from CM";
+        validateStackStatus(stack, operationDescription);
+        validateInstanceStatus(stack, operationDescription);
 
         LOGGER.debug("Triggering sync from CM to db: syncing versions from CM to db, nameOrCrn: {}, workspaceId: {}, candidateImageUuids: {}",
                 nameOrCrn, workspaceId, candidateImageUuids);
         return syncComponentVersionsFromCm(stack, candidateImageUuids);
     }
 
+    private void validateInstanceStatus(Stack stack, String operationDescription) {
+        if (instanceMetaDataService.anyInstanceStopped(stack.getId())) {
+            String message = String.format("Please start all stopped instances. %s can only be made when all your nodes in running state.",
+                    operationDescription);
+            LOGGER.debug(message);
+            throw new BadRequestException(message);
+        }
+    }
+
+    private void validateStackStatus(Stack stack, String operationDescription) {
+        if (stack.getStackStatus().getStatus().isStopState()) {
+            String message = String.format("%s cannot be initiated as the cluster is in %s state.",
+                    operationDescription, stack.getStackStatus().getStatus());
+            LOGGER.debug(message);
+            throw new BadRequestException(message);
+        }
+    }
```

Comment:
I am not sure about which use case will call these, but we need more detailed method names probably, because if someone will use these or these are used during some usecase of "stop/start scaling"/"fast scaling", it will fail, since cluster will have stopped nodes most probably

Reply:
discussed: ensureInstanceAvailability and ensureStackAvailability